environment/disaster_recovery/ecs.tf

```diff
@@ -0,0 +1,148 @@
+        awslogs-region        = "eu-west-1",
+        awslogs-stream-prefix = "dr-backup-${var.environment}"
+      }
+    },
+    environment = [
+      {
+        name  = "KMS_KEY_ID",
+        value = aws_kms_key.db_backup.id
+      },
+      {
+        name  = "BACKUP_ACCOUNT",
+        value = var.backup_account_id
+      },
+      {
+        name  = "SOURCE_ACCOUNT",
+        value = var.account.account_id
+      },
+      {
+        name  = "BACKUP_ACCOUNT_ROLE",
+        value = aws_iam_role.cross_acc_backup.arn
+      },
+      {
+        name  = "DB_ID"
+        value = "${var.db.name}-${var.environment}"
+      },
+    ]
+  })
+}
+
+resource "aws_cloudwatch_log_group" "dr_backup" {
+  name              = "dr-backup-${var.environment}"
+  retention_in_days = var.log_retention
+  tags              = var.default_tags
+}
+
+//TASK ROLE
+
+resource "aws_iam_role" "dr_backup" {
+  assume_role_policy = var.task_role_assume_policy.json
+  name               = "dr-backup.${var.environment}"
+  tags               = var.default_tags
+}
+
+data "aws_iam_policy_document" "dr_backup" {
+  statement {
+    sid       = "allowAssumeAccess"
+    effect    = "Allow"
+    resources = [aws_iam_role.cross_acc_backup.arn]
+    actions = [
+      "sts:AssumeRole"
+    ]
+  }
+  statement {
+    sid    = "allowKMSAccess"
+    effect = "Allow"
+    actions = [
+      "kms:CreateGrant",
+      "kms:DescribeKey"
+    ]
+    resources = [
+      "*"
+    ]
+  }
+  statement {
+    sid       = "allowSnapshotAccess"
+    effect    = "Allow"
+    resources = ["*"]
+    actions = [
+      "rds:CreateClusterDBSnapshot",
+      "rds:CreateDBSnapshot",
+      "rds:DeleteDBClusterSnapshot",
+      "rds:DeleteDBSnapshot",
+      "rds:DescribeDBInstances",
+      "rds:DescribeDBClusters",
+      "rds:DescribeDBClusterSnapshots",
+      "rds:DescribeDBSnapshots",
+      "rds:ModifyDBClusterSnapshotAttribute",
+      "rds:ModifyDBSnapshotAttribute",
+      "rds:DescribeDBClusterSnapshotAttributes",
+      "rds:DescribeDBSnapshotAttributes",
+      "rds:CopyDBClusterSnapshot",
+      "rds:CopyDBSnapshot",
+      "rds:ListTagsForResource"
+    ]
+  }
+}
+
+resource "aws_iam_role_policy" "dr_backup" {
+  name   = "dr-backup-task.${var.environment}"
+  policy = data.aws_iam_policy_document.dr_backup.json
+  role   = aws_iam_role.dr_backup.id
+}
+
+locals {
+  dr_backup_sg_rules = {
+    ecr     = var.common_sg_rules.ecr
+    logs    = var.common_sg_rules.logs
+    s3      = var.common_sg_rules.s3
+    ssm     = var.common_sg_rules.ssm
+    ecr_api = var.common_sg_rules.ecr_api
+    dr_backup = {
+      port        = 443
+      type        = "egress"
+      protocol    = "tcp"
+      target_type = "cidr_block"
+      target      = "0.0.0.0/0"
+    }
+  }
+}
+
+module "dr_backup_security_group" {
+  source = "../security_group"
+  rules  = local.dr_backup_sg_rules
+  name   = "dr-backup"
+  tags   = var.default_tags
+  vpc_id = var.aws_vpc_id
+}
```

Comment:
Don't know if this is possible in modules but we've tended to follow the pattern of extracting SGs into their own file for ease of update.

Reply:
Yeah put in separate file. Good point